sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -93,6 +93,28 @@ object IntervalUtils {
 
   private val yearMonthPattern = "^([+|-])?(\\d+)-(\\d+)$".r
 
+  private val yearMonthFuzzyPattern = "[^-|+]*?([+|-]?[\\d]+-[\\d]+).*".r
+
+  def safeFromYearMonthString(input: UTF8String): Option[Int] = {
```

Comment:
Why not just call `fromYearMonthString`?

Reply:
> Why not just call `fromYearMonthString`?

Null safe? like `safeStringToInterval` we can avoid NPE

Reply:
It's a new feature, we can use the ANSI behavior directly.

Reply:
> It's a new feature, we can use the ANSI behavior directly.

So just let it throw exception? Update later

Reply:
How about current?